Add NumericField Control

IIARROWS opened the pull request:
Added new control `MudNumericField` with Material UI up/down spin buttons.

See related closed PR: https://github.com/Garderoben/MudBlazor/pull/843

tungi52 committed:
Allow drawers to use different sizes

IIARROWS committed:
Add NumericField Control

IIARROWS committed:
Add NumericField Control

henon commented:
Cool, you squashed it. Can you add some tests? Take a look at the TextField tests, should be easy to do something similar with this component.

mikes-gh commented on src/MudBlazor/MudBlazor.csproj:
```diff
@@ -35,6 +35,7 @@
   <ItemGroup>
     <None Include="..\..\LICENSE" Pack="true" Visible="false" PackagePath="" />
     <None Include="..\..\content\Nuget.png" Pack="true" Visible="false" PackagePath="" />
+    <None Include="Components\NumericField\MudNumericField.razor" />
```
Please could you remove this project file change

IIARROWS replied:
Done with commit https://github.com/Garderoben/MudBlazor/pull/1000/commits/9c5c4e84c92c8c0f9a6579877c7efcb15b587dab

IIARROWS committed:
Removed new file reference from MudBlazor.csproj

IIARROWS committed:
Removed new file reference from MudBlazor.csproj

mikes-gh commented on src/MudBlazor/Styles/components/_input.scss:
```diff
@@ -297,152 +297,161 @@
 }
 
```
Any chance you could undo the refactoring in this file.  As you can see from the diff below its hard to review the changes which could potentially affect other component 👍

IIARROWS replied:
I'm not doing anything...

IIARROWS replied:
Besides, it seems to just align the formatting to the entire file, as some parts have 2 spaces and others 4*, but for whatever reason it doesn't say it changed.

* And that's the reason why tabs are the only way to format a document.

mikes-gh replied:
You've probably got VS auto-formatting.  Which when working by yourself is good but for PR it makes it harder

mikes-gh replied:
BTW the formatting is 2 spaces so VS is doing the right thing just in this case we probably better do it a s a seperate task 👍

IIARROWS committed:
Minor MudNumericField css class for spinner buttons

IIARROWS committed:
Minor MudNumericField css class for spinner buttons

IIARROWS committed:
Minor MudNumericField css class for spinner buttons

IIARROWS committed:
Minor MudNumericField css class for spinner buttons

IIARROWS committed:
Merge branch 'feature/numeric-field-control' of https://github.com/IIARROWS/MudBlazor into feature/numeric-field-control

Commit:
Stop exceptions on page leave for tabs

Commit:
Fix logo 404

Garderoben committed:
MudSwitch added full color support #1064

Garderoben committed:
Docs: Removed old/broken icons from TreeView Example.

Commit:
Fix TreeviewCustomExample

Commit:
Fix TreeViewItemTemplateExample

Garderoben committed:
Updated Badge example Google Mobile Usability Warning #1069

tungi52 committed:
Update DefaultTheme.razor

TyrenDe committed:
Fix the issue with server side paging after changing page size

tungi52 committed:
Scrolling tabs on panel activation

grishat committed:
FIX: README.md

Commit:
Update table docs Breakpoint.None

tungi52 committed:
fix hidden rating star inside a dialog

grishat committed:
ADD: ToolBar example into MudBlazor.Docs

Garderoben committed:
MudCharts: Simplified Line code a little

Garderoben committed:
MudCharts added bar chart

Garderoben committed:
Docs: Added Bar chart page and example

tungi52 committed:
Fix mudcollapse default open state

IIARROWS committed:
Automation tests NumericField and fix tab stopping on spinners

IIARROWS committed:
Automation tests NumericField and fix tab stopping on spinners

Commit:
Fix mediatype for DialogExample

Commit:
Fix some 404s and disallow russian bot

Commit:
Fix some 404s and disallow russian bot

porkopek committed:
Fix mud-picker-inline width in Grid

tungi52 committed:
fix table multiselect excessivecallback (#1104)

TyrenDe committed:
Add date format changed hook for descendants (#1100)

* Add date format changed hook for descendants

* Apply feedback to always call format changed callback

tungi52 committed:
fix MudDateRangePicker initial render

Commit:
Change js script order

Commit:
Change js script order

Garderoben committed:
Docs: Changed Garderobens & Henons broken linked in urls to gravatar

Garderoben committed:
Docs: Fixed layout on Alert Elevation example on mobiles #1083

Garderoben committed:
Docs: Fixed Mobile Layout on Element page #1083

Garderoben committed:
Docs: Fixed Mobile Layout on Overlay Loader example #1083

Garderoben committed:
Docs: Fixed Mobile Layout on Mud Icons page #1083

Commit:
Fix incorrect check for missing js file

Garderoben committed:
Docs Server: Removed backslash in Host.cshtml

porkopek committed:
Changed: Pickers are not inlined and have 100 width

Commit:
dotnet format

porkopek committed:
Update _picker.scss

Garderoben committed:
Docs: Fixed spelling error on Display page.

Garderoben committed:
Docs: Fixed broken link on MudHidden page.

Commit:
Cache bust 5.0.4

Garderoben committed:
Docs: Removed MudHidden from MainLayout and changed it CSS classes.

Garderoben committed:
Docs: Fixed height of divider in appbar

Garderoben committed:
MudTable cell width/alignment fixed #1115

Garderoben committed:
Docs: Updated TableBasicExample

Garderoben committed:
Updated to version 5.0.4

Garderoben committed:
Docs: Updated footer to version 5.0.4

Garderoben committed:
Docs: TableBasicExample removed unused bool.

Commit:
Merge branch 'dev'

Commit:
Merge branch 'master' into dev

tungi52 committed:
add action buttons to pickers

ChocoboCoder committed:
Update DoNotRemove.txt

Garderoben committed:
Fixed PR#1099 MudPickerInline TextField

Garderoben committed:
Docs: Fixed MudPicker examples.

Garderoben committed:
Added missing Style and Class option to MudTabPanel #1130

Garderoben committed:
Updated to 5.0.5

Garderoben committed:
Updated to 5.0.5

Garderoben committed:
Cache bust 5.0.5

Garderoben committed:
Made Mony smaller.

tungi52 committed:
change drawer-content height to 100%

tungi52 committed:
add clear option to pickers

Commit:
Fix #1103 Arithmetic overflow in chart examples

botwinter committed:
Add support for Required parameter to Select component (#1127)

tungi52 committed:
Add key to MudDialogInstance (#1137)

Commit:
Fix #1144 Scroll to top example

Garderoben committed:
Added some base position/visibility classes.

Garderoben committed:
Docs: ScrollToTop removed inline <style> css from examples.

Garderoben committed:
Removed svg height from base scss.

IIARROWS committed:
Fixed NumericField debouncing,
Update appropriate numeric types for NumericField examples.

IIARROWS committed:
Fixed NumericField debouncing,
Update appropriate numeric types for NumericField examples.

tungi52 committed:
fix #1153

tungi52 committed:
fix #994

Garderoben committed:
Docs: Removed docs specific component.

Commit:
Add application insights to website

Commit:
Add application insights to website

tungi52 committed:
fix #1169

Commit:
Add test today should be selected

tungi52 committed:
Fire onclick event in MudNavLink

tungi52 committed:
fix appbar/maincontent css in firefox esr

Commit:
fix spelling error

Commit:
minor grammar fixes

Commit:
improve title grammar

Commit:
include subtitle  and description properties for the header

Commit:
add information regarding material design surfaces

Commit:
minor grammar and punctuation fixes

Commit:
Fix flaky async autocomplete test

ChrTall committed:
Fixed Typo in FlexPage.razor

Garderoben committed:
MudSwitch removed Class from the internal switch class, Fix for #1163

ClumsyPenguin committed:
Interpolation for line charts (#1139)

* added interpolation mechanism

* commit to continue work on laptop

* Cache bust 5.0.5

* first interpolation implementation

* spliced code up to reuse and add in documentation

* Add support for Required parameter to Select component (#1127)

* Add key to MudDialogInstance (#1137)

* Fix #1103 Arithmetic overflow in chart examples

* Fix #1144 Scroll to top example

* fixed initial Y value

* made splined width 100%

* added menu for all the different interpolation algorithms

* Added some base position/visibility classes.

* big cleanup

* cleanup notinterpolation constructor

* Docs: ScrollToTop removed inline <style> css from examples.

* quickly added latest suggestions

* Removed svg height from base scss.

* add action buttons to pickers

* add clear option to pickers

* renamed curveEnum to Interpolationoptions

* change drawer-content height to 100%

* fix #1153

* fix #994

* Docs: Removed docs specific component.

* Add application insights to website

* Add application insights to website

* fix #1169

* Add test today should be selected

* fix appbar/maincontent css in firefox esr

* Fix flaky async autocomplete test

* Fire onclick event in MudNavLink

* fix spelling error

* minor grammar fixes

* improve title grammar

* include subtitle  and description properties for the header

* add information regarding material design surfaces

* minor grammar and punctuation fixes

* Fixed Typo in FlexPage.razor

* MudSwitch removed Class from the internal switch class, Fix for #1163

Commit:
Fix dispose DotNetObjectReference

Commit:
Fix dispose DotNetObjectReference

Commit:
EditForm support: respect data validation attributes on properties targeted by `For` (#1198)

* feat: add support of data validation attributes on properties targeted by `For`

* refactor: apply feedback

See https://github.com/Garderoben/MudBlazor/pull/1198#discussion_r594914599

Commit:
EditForm support: respect data validation attributes on properties targeted by `For` (#1198)

* feat: add support of data validation attributes on properties targeted by `For`

* refactor: apply feedback

See https://github.com/Garderoben/MudBlazor/pull/1198#discussion_r594914599

(cherry picked from commit 0f6d5ae9e9cc0fa724f5b84e5257f706e3cb6ce1)

Commit:
Don't raise exception if elementRef null

Commit:
Fix object disposed exception

Garderoben committed:
Remove package reference.

Garderoben committed:
Merge remote-tracking branch 'origin/feature/charts' into dev

Commit:
Tidy-up interpolated charts

Commit:
Correct pluralisation of Series

Garderoben committed:
Updated to version 5.0.6

Commit:
Cache bust 5.0.6

Commit:
Merge branch 'dev'

henon committed:
Form: fixed bug introduced by PR #1198 + more tests

henon committed:
fixed warning

Commit:
Treat warnings as errors for all builds

tungi52 committed:
Add dense mode to pickers

tungi52 committed:
breadcrumb-separator dark mode

tungi52 committed:
Add disabled property for MudLink

GerkinDev committed:
MudForm: when using For param, even w/o edit form, retrieve error messages from data attributes (#1212)

* feat: use data attributes even if no edit form, retrieve error messages

* fix: remove nullable where it caused error in test pipeline

See See https://github.com/Garderoben/MudBlazor/pull/1212/checks?check_run_id=2129603317

* fix: use `EditContext.Model` if defined

Note: Use of cross referencing validation attributes (like `CompareAttribute`) requires an EditContext. It would be great if the model could be bound directly on the `MudForm` component as an additional parameter.

* feat: add ValidationAttribute exception's message to errors list

Application of suggestion https://github.com/Garderoben/MudBlazor/pull/1212#issuecomment-801024103

Commit:
Fix flaky tests

henon committed:
Input/TextField/Autocomplete: fix swallowing of input while typing with high latency

henon committed:
Fixed failing test (had to manually call Blur() to simulate real conditions)

henon committed:
minor change

(cherry picked from commit a63dba244b485e3240437bcc2c435d588d089dea)

GerkinDev committed:
test: add test to ensure ValidationContext is set everytime.

Related to Garderoben/MudBlazor#1229

(cherry picked from commit f63dcf0748ab8d1ee0a35e2e333e9bd343000f1f)

IIARROWS committed:
Add KeyXPreventDefault on MudInput

Garderoben committed:
Docs: Fixed responsive issue on Table Page. #1230

Garderoben committed:
Cleaned up slider scss and added color support for full pallet #1236

Garderoben committed:
Docs: Added different colors on the slider examples.

Garderoben committed:
Docs: Added favicon/icon for IOS, Android, MS

Garderoben committed:
Changed GitHub Banned to transparent version.

IIARROWS committed:
Add Prevent Default

Int32Overflow commented:
I don't want to sound impatient, but do you know when it will be finished?

IIARROWS committed:
Merge branch 'feature/numeric-field-control' of https://github.com/IIARROWS/MudBlazor into feature/numeric-field-control

IIARROWS commented:
I have no idea how did 133 commits got here... I just rebased the branch and add one commit.

henon commented:
That's becaise of the merge.  Had you rebased instead of merging they would not be visible here. Once there is a merge commit in the branch rebasing doesn't help any more. 

But don't worry, it won't matter when this is squash merged.

IIARROWS commented:
That's exactly what I did, the only merges are the one on the branch I refused from.

Int32Overflow commented:
Can I support you with this issue? What is missing?

henon commented:
Garderoben and I finally decided to merge it. Sorry for the long delay, we were really busy lately.
Many thanks @IIARROWS !

Int32Overflow commented:
No problem. Thank you very much @henon and @IIARROWS

henon commented:
> I have no idea how did 133 commits got here... I just rebased the branch and add one commit. And I couldn't squash it.

At the time I had no idea what the reason for this phenomenon was. We now know that merging/rebasing with Visual Studio's git support is the culprit for these problems as every time it happened the user used VS to merge or rebase. Last time it happened we found out that the user accidentally rebased the dev branch onto their PR branch instead of the other way around. VS seems to make it very easy to get this wrong.

For the record, neither a simple merge of the dev branch into the PR branch nor rebasing the PR branch onto the dev branch using a decent git tool will cause unrelated commits to show up in the PR.
